pkg/federatedtypes/service.go

```diff
@@ -92,6 +92,28 @@ func (a *FederatedServiceAdapter) ObjectForCluster(template, override pkgruntime
 	return service
 }
 
+// TODO(shashi) Avoid the need for this adapter method by handling the scenario generically
+// for cases where the field in spec is updated by controllers instead of status.
+func (a *FederatedServiceAdapter) ObjectForUpdateOp(desiredObj, clusterObj pkgruntime.Object) pkgruntime.Object {
+	desiredService := desiredObj.(*corev1.Service)
+	clusterService := clusterObj.(*corev1.Service)
+
+	// ClusterIP and NodePort are allocated to Service by cluster, so retain the same if any while updating
+	desiredService.Spec.ClusterIP = clusterService.Spec.ClusterIP
+	for i, fPort := range desiredService.Spec.Ports {
```

Comment:
(No action required) Maybe choose a more descriptive name than fPort?  I parse cPort as c[luster]Port, but I don't know what the `f` stands for.